src/cmake/__init__.py

```diff
@@ -2,7 +2,7 @@
 import subprocess
 import sys
 
-if sys.version_info < (3, 10):
+if sys.version_info < (3, 8):
     from importlib_metadata import distribution
 else:
     from importlib.metadata import distribution
```

Comment:
I don't think `importlib.metadata.distribution` is available with Python < 3.10

| Python Version  | `importlib.metadata` | URL |
| --|--|--|
| 3.8  | :x: | https://github.com/python/cpython/tree/3.8/Lib/importlib |
| 3.9  | :x: | https://github.com/python/cpython/tree/3.9/Lib/importlib |
| 3.10 | :white_check_mark:  | https://github.com/python/cpython/tree/3.10/Lib/importlib|

Reply:
I then suggest to keep the original patch

Reply:
* https://github.com/python/cpython/blob/95c340a86b828cac6c0a2e4f2fa8a96695388c73/Lib/importlib/metadata.py#L27
* https://github.com/python/cpython/blob/95c340a86b828cac6c0a2e4f2fa8a96695388c73/Lib/importlib/metadata.py#L27

?

Reply:
It is:

```
^[[Amathias@wampa:~$ docker run --rm -it python:3.8 python3 -c 'from importlib_metadata import distribution'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'importlib_metadata'
mathias@wampa:~$ docker run --rm -it python:3.8 python3 -c 'from importlib.metadata import distribution'
```

Reply:
https://github.com/python/cpython/blob/95c340a86b828cac6c0a2e4f2fa8a96695388c73/Lib/importlib/metadata.py#L497

Reply:
(I manually checked this on 3.8 before https://github.com/scikit-build/cmake-python-distributions/pull/474#issuecomment-2020805647 after checking the docs)

```PyCon
$ python3.8                                                                                                                        
>>> import importlib.metadata
>>> importlib.metadata.distribution("pip").files[0].locate().resolve()
PosixPath('/usr/local/lib/python3.8/site-packages/AUTHORS.txt')
```

Reply:
Thanks for clarifying, not seeing the "metadata" directory tricked me.